### PR TITLE
chore: revert to WorkflowEntity

### PR DESCRIPTION
**Pull request by octonato**
Draft because is depending on a new proxy snapshot release (see https://github.com/lightbend/kalix-proxy/pull/1849).

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ object Dependencies {
   object Kalix {
     val ProtocolVersionMajor = 1
     val ProtocolVersionMinor = 0
-    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-85ad856")
+    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-13-a6ec96a7-SNAPSHOT")
```
Comment (octonato) [inline project/Dependencies.scala:9]:
```suggestion
    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-85ad856")
```

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala [hunk 2]
```diff
@@ -38,12 +38,12 @@ import kalix.javasdk.impl.workflow.WorkflowEffectImpl.TransitionalEffectImpl
 import kalix.javasdk.impl.workflow.WorkflowEffectImpl.UpdateState
```
Comment (aludwiko) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala:38]: WorkflowEntityImpl?

Comment (aludwiko) [reply sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala:38]: if yes, then I suppose we need to change also other classes in `/impl/workflow` package, and actually rename the package as well?

File: sdk/spring-sdk/src/main/java/kalix/springsdk/workflow/ReflectiveWorkflowProvider.java [hunk 3]
```diff
@@ -19,8 +19,8 @@
 import com.google.protobuf.Descriptors;
```
Comment (aludwiko) [inline sdk/spring-sdk/src/main/java/kalix/springsdk/workflow/ReflectiveWorkflowProvider.java:19]: the same story with renaming a package with classes

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala (hunk 2 above)
Comment (octonato) [reply sdk/java-sdk/src/main/scala/kalix/javasdk/impl/workflow/WorkflowImpl.scala:38]: yep, I will do it.

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/EntityExceptions.scala [hunk 4]
```diff
@@ -76,6 +77,9 @@ object EntityExceptions {
 
     def apply(init: ReplicatedEntityInit, message: String): EntityException =
       ProtocolException(init.entityId, message)
+
+    def apply(init: WorkflowInit, message: String): EntityException =
```
Comment (octonato) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/EntityExceptions.scala:81]: we may want to change the protocol once again to align here. `WorkflowInit` should become `WorkflowEntityInit`

File: project/Dependencies.scala [hunk 5]
```diff
@@ -6,7 +6,7 @@ object Dependencies {
   object Kalix {
     val ProtocolVersionMajor = 1
     val ProtocolVersionMinor = 0
-    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-85ad856")
+    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-14-2ec41f63-SNAPSHOT")
```
Comment (octonato) [inline project/Dependencies.scala:9]:
```suggestion
    val ProxyVersion = System.getProperty("kalix-proxy.version", "1.0.31-14-f8e34156")
```

**Review by aludwiko — APPROVED** [review]

**Pull request merged**